### ISSUE TITLE
Table of Contents not visible in blog with viewports less than 960px in width #5975

### ISSUE THREAD

**Issue by LandonTClipp** [squidfunk/mkdocs-material#6084]
### Bug description

In the official `Blog Support Just Landed` blog post, you can see when the viewport is less than 960 pixels in width, there is no way to view the TOC for the post itself.

![Screenshot 2023-09-25 at 5 20 17 PM](https://github.com/squidfunk/mkdocs-material/assets/11232769/780894e5-2d48-4107-b215-4f602fa49dae)

Attached is a minimal reproducible example using [this repo](https://github.com/LandonTClipp/mkdocs-material-blog-toc-bug).

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
- https://github.com/squidfunk/mkdocs-material/discussions/5975
- https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/?h=blog#configuration
- https://github.com/LandonTClipp/mkdocs-material-blog-toc-bug

### Reproduction

[9.4.2-blog-toc-missing.zip](https://github.com/squidfunk/mkdocs-material/files/12719838/9.4.2-blog-toc-missing.zip)

### Steps to reproduce

1. Create new blog
2. Create a single post with a couple of markdown sections. Add categories to the post.
3. View the post in the browser. Reduce viewport to <960px
4. Observe there is no TOC for the post itself, and no way to drill down into the TOC as is typical with non-blog pages.

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for sharing. As noted #5975, it's a current design limitation, as there's no active page that we can use to attach the table of contents to in the sidebar, but we'll check if we can make it happen.

**Comment by LandonTClipp** [comment]
I was thinking about this the other day. My view on this is that it would be sufficient if the sidebar noted here in the screenshot first contained a TOC of the current post being viewed, and below the TOC would be the blog categories. People are going to be interacting with the TOC far more than the blog categories so it would make sense to prioritize that in the sidebar view.

**Comment by squidfunk** [comment]
I agree, but as said, it's not simple given how navigation works. It is exceptionally tricky.

**Comment by squidfunk** [comment]
We're not considering trying to try and fix this at this point in time. The reason is that we would need to add exceptions to the way navigation is handled, which is basically all-CSS. We've learned a lot in the last 7 years since we adopted the current navigation, and will revise it when we do the next face lift, which we're planning to do later this year. I can't put an estimate on it, given how many other things we currently need to do, but it will happen. In this instance, we'll also revisit this problem.

Until then, we're closing it as won't fix, but as said, it might be fixable with a major redesign.
